src/common/snippets/include/snippets/pass/extract_reshapes_from_mha.hpp

```diff
@@ -29,10 +30,46 @@ namespace pass {
  *
  * @ingroup snippets
  */
-class ExtractReshapesFromMHA: public ov::pass::MatcherPass {
+class ExtractPairsAfterMatmul : public ov::pass::MatcherPass {
 public:
-    OPENVINO_MATCHER_PASS_RTTI("snippets::pass::ExtractReshapesFromMHA");
-    ExtractReshapesFromMHA();
+    OPENVINO_MATCHER_PASS_RTTI("snippets::pass::ExtractPairsAfterMatmul");
+    ExtractPairsAfterMatmul();
+};
+
+/**
+ * @interface RankUpgradeToRankReduction
```

Comment:
I am wondering if we can apply this transformation to partially dynamic shape... An original target model, for which the transformation is introduced, has dynamic shapes. Let's discuss offline if the pass can be improved to cover broader cases

Reply:
Discsussed offline, currently only static is the focus.